test/system/src/test/java/io/pravega/test/system/ReaderCheckpointWithInternalThreadpool.java

```diff
@@ -0,0 +1,246 @@
+import io.pravega.client.stream.impl.JavaSerializer;
+import io.pravega.common.concurrent.ExecutorServiceHelpers;
+import io.pravega.common.concurrent.Futures;
+import io.pravega.test.system.framework.Environment;
+import io.pravega.test.system.framework.SystemTestRunner;
+import io.pravega.test.system.framework.Utils;
+import io.pravega.test.system.framework.services.Service;
+import lombok.Cleanup;
+import lombok.extern.slf4j.Slf4j;
+import mesosphere.marathon.client.MarathonException;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.junit.runner.RunWith;
+import java.io.Serializable;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+import static io.pravega.test.system.AbstractSystemTest.*;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+@Slf4j
+@RunWith(SystemTestRunner.class)
+public class ReaderCheckpointWithInternalThreadpool {
+    private static final long READ_TIMEOUT = SECONDS.toMillis(500);
+    private static final String SCOPE = "scope";
+    private static final String STREAM = "checkpointStream";
+    private static final String READER_GROUP_NAME = "checkpointRG";
+    private static final String CHECKPOINT1 = "checkpoint1";
+    private static final String CHECKPOINT2 = "checkpoint2";
+    private static final String READER = "reader";
+    @Rule
+    public Timeout globalTimeout = Timeout.seconds(7 * 60);
+    private URI controllerURI = null;
+    private StreamManager streamManager = null;
+    private Controller controller = null;
+    private ConnectionFactory connectionFactory = null;
+    private final StreamConfiguration streamConfig = StreamConfiguration.builder()
+            .scalingPolicy(ScalingPolicy.fixed(1)).build();
+
+    private final ScheduledExecutorService readerExecutor = ExecutorServiceHelpers.newScheduledThreadPool(1,
+            "readerCheckpoint-reader");
+    
+    @Environment
+    public static void initialize() throws MarathonException, ExecutionException {
+        URI zkUri = startZookeeperInstance();
+        startBookkeeperInstances(zkUri);
+        URI controllerUri = ensureControllerRunning(zkUri);
+        ensureSegmentStoreRunning(zkUri, controllerUri);
+    }
+
+    @Before
+    public void setup() {
+        controllerURI = fetchControllerURI();
+        streamManager = StreamManager.create(Utils.buildClientConfig(controllerURI));
+
+        connectionFactory = new SocketConnectionFactoryImpl(ClientConfig.builder().build());
```

Comment:
Here two different client configs are being used. And a duplicate one is also created on 111

Reply:
done